File: Easy/Valid_Palindrome/Valid_Palindrome.js
class Solution {
    isAlphaNumeric(c) {
        return (
            (c >= "A" && c <= "Z") ||
            (c >= "a" && c <= "z") ||
            (c >= "0" && c <= "9")
        )
    }

    /**
     * @param {string} s
     * @return {boolean}
     */
    isPalindrome(s) {
        let i = 0;
        let j = s.length - 1;
        while (i < j) {
            while (i < j && !this.isAlphaNumeric(s[i])) {
                i++;
            }
            while (j > i && !this.isAlphaNumeric(s[j])) {
                j--;
            }
            if (s[i].toLowerCase() !== s[j].toLowerCase()) {
                return false;
            }
            i++;
            j--;
        }

        return true;
    }
}

File: Easy/Valid_Palindrome/Valid_Palindrome.java
class Solution {

    public boolean isAlphaNumeric(char c) {
        return (
            (c >= 'A' && c <= 'Z') ||
            (c >= 'a' && c <= 'z') ||
            (c >= '0' && c <= '9')
        );
    }

    public boolean isPalindrome(String s) {
        int i = 0;
        int j = s.length() - 1;

        while (i < j) {
            while (i < j && !this.isAlphaNumeric(s.charAt(i))) {
                i++;
            }
            while (j > i && !this.isAlphaNumeric(s.charAt(j))) {
                j--;
            }
            if (Character.toLowerCase(s.charAt(i)) != Character.toLowerCase(s.charAt(j))) {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}


